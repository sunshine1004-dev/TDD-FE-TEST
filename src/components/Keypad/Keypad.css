.keypad-container {
  display: flex;
  flex-direction: row;
  padding: 2%;
  margin-top: 0.5rem;
}
.argument-input {
  box-shadow: 0 0 0.25rem rgba(0, 0, 0, 0.15);
  border: 0.125rem solid #cccbcd;
  border-radius: 0.5rem;
  color: #3d3c3e;
}

.argument-input:focus {
  border-color: #5f2ad0;
  outline: none;
  caret-color: #5f2ad0;
}

.submit-button {
  cursor: pointer;
  border-radius: 0.5rem;
  border: 0.125rem solid currentColor;
  background: #fff;
  color: #124ce1cc;
  height: 2.2rem;
  width: 7rem;

  margin-left: 0.5rem;
}

.submit-button:hover {
  transition: all 0.2s ease 0s;
  box-shadow: rgb(4 17 29 / 25%) 0px 0px 8px 0px;
  background: #124ce1cc;
  color: #fff;
}
