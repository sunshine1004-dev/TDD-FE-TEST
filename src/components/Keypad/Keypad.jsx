import React from 'react';
import PropTypes from 'prop-types';
import './Keypad.css';

const Keypad = ({ argument, setArguments, handleSubmit }) => {
    return (
        <div className="keypad-container">
            <input
                className="argument-input"
                autoFocus
                onChange={(e) => setArguments(e.target.value)}
                value={argument}
                placeholder="1,3,5"
            />
            <button className="submit-button" onClick={handleSubmit}>
                submit
            </button>
        </div>
    );
};

Keypad.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    setArguments: PropTypes.func.isRequired,
};

export default Keypad;
