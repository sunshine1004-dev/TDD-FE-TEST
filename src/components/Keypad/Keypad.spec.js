import React from 'react';
import { shallow } from 'enzyme';
import Keypad from './Keypad';

const MOCK_SET_ARGUMENTS = jest.fn();
const MOCK_HANDLE_SUBMIT = jest.fn();
const MOCK_ARGUMENT = '1,3,5';

describe('Keypad', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Keypad
        argument={MOCK_ARGUMENT}
        setArguments={MOCK_SET_ARGUMENTS}
        handleSubmit={MOCK_HANDLE_SUBMIT}
      />
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render 1 <input />', () => {
    expect(wrapper.find('input').length).toEqual(1);
  });

  it('should render 1 <button />', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });
});
