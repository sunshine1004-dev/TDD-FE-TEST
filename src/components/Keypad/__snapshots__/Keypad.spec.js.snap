// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render correctly 1`] = `
<div
  className="keypad-container"
>
  <input
    autoFocus={true}
    className="argument-input"
    onChange={[Function]}
    placeholder="1,3,5"
    value="1,3,5"
  />
  <button
    className="submit-button"
    onClick={[MockFunction]}
  >
    submit
  </button>
</div>
`;
